libs/@guardian/source/src/react-components/utils/colourConversions.ts

```diff
@@ -0,0 +1,139 @@
+/*
+	Implementation from https://github.com/Qix-/color-convert
```

Comment:
@SiAdcock do I need to do anything special with the license?

Reply:
Oh good point! It's MIT, so pretty permissive as long as we [include the licence in the code](https://snyk.io/articles/what-is-mit-license/). Shall we copy-paste [the licence](https://github.com/Qix-/color-convert?tab=License-1-ov-file) into a comment in this file?